internal/services/digitaltwins/digital_twins_instance_resource.go

```diff
@@ -149,10 +149,15 @@ func resourceDigitalTwinsInstanceUpdate(d *pluginsdk.ResourceData, meta interfac
 		props.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
 	}
 
-	if _, err := client.Update(ctx, id.ResourceGroup, id.Name, props); err != nil {
+	future, err := client.Update(ctx, id.ResourceGroup, id.Name, props)
+	if err != nil {
 		return fmt.Errorf("updating Digital Twins Instance %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
 
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for update of the Digital Twins Instance %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
```

Comment:
These log messages could be simplified a bit
```suggestion
		return fmt.Errorf("updating %s: %+v", *id, err)
	}

	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
		return fmt.Errorf("waiting for update of %s: %+v", *id, err)
```